checks/branch_protection.go

```diff
@@ -28,36 +28,56 @@ type branchProtectionSetting int
 
 const (
 	// CheckBranchProtection is the exported name for Branch-Protected check.
-	CheckBranchProtection                         = "Branch-Protection"
-	minReviews                                    = 2
-	allowForcePushes      branchProtectionSetting = iota
+	CheckBranchProtection = "Branch-Protection"
+	minReviews            = 2
+	// Points incremented at each level.
+	level1 = 3
+	level2 = 3
+	level3 = 2
+	level4 = 1
+	level5 = 1
+	// First level.
+	allowForcePushes branchProtectionSetting = iota
 	allowDeletions
-	requireLinearHistory
+	// Second and third level.
+	requireApprovingReviewCount
+	// Fourth level.
+	requireStatusChecksContexts
+	// Admin settings.
+	// First level.
 	enforceAdmins
+	// Second level.
 	requireUpToDateBeforeMerge
-	requireStatusChecks
-	requireStatusChecksContexts
-	requireApprovingReviewCount
+	// Fifth level.
 	dismissStaleReviews
-	requireCodeOwnerReviews
+	// requireCodeOwnerReviews no longer used.
+	// requireLinearHistory no longer used, see https://github.com/ossf/scorecard/issues/1027.
 )
 
 var branchProtectionSettingScores = map[branchProtectionSetting]int{
-	allowForcePushes:     1,
-	allowDeletions:       1,
-	requireLinearHistory: 1,
-	// Need admin token.
-	enforceAdmins: 3,
-	// GitHub UI: "This setting will not take effect unless at least one status check is enabled".
-	// Need admin token.
-	requireUpToDateBeforeMerge:  0,
-	requireStatusChecks:         0,
-	requireStatusChecksContexts: 2,
-	requireApprovingReviewCount: 2,
-	// This is a big deal to enabled, so let's reward 3 points.
-	// Need admin token.
-	dismissStaleReviews:     3,
-	requireCodeOwnerReviews: 2,
+	allowForcePushes:            1,
+	allowDeletions:              1,
+	requireStatusChecksContexts: 1, // Gated by requireStatusChecks=true.
+	requireApprovingReviewCount: 1,
+	// Need admin token, so we cannot rely on them in general.
+	enforceAdmins:              1,
+	requireUpToDateBeforeMerge: 1, // Gated by requireStatusChecks=true.
```

Comment:
I don't think this one is gated, is it? the scorecard check will use the settings I think because of this comment ("Even though it technically is not enforced by GitHub if Context=nil, we still show a positive outcome for users.")

Reply:
right. When I tested, I think this setting was always nil when Context=nil. So I think my comment no longer applies. Let me remove it.